### DISCUSSION TITLE
full featured table widget

### DISCUSSION THREAD

**Discussion by aforsythe** [marimo-team/marimo#1663] [Ideas]
It would be great to have a more full featured table widget like [rhandsontable](https://jrowen.github.io/rhandsontable/)

**Comment by akshayka** [comment]
@aforsythe 

We've recently added more features to `ui.table`, including global search, column filters that can be stacked (with logical AND), and sorting by columns.

What additional features would you like? cc @mscolnick

**Reply by aforsythe** [reply]
Thanks @akshayka 

The most important feature for me would be the ability to edit tables directly specifically with the ability to paste data into the tables from excel. That would allow for easy data input when used in a Marimo web app. See http://ssi-calculator.oscars.org

**Comment by aforsythe** [comment]
@akshayka any thoughts on this?

**Reply by akshayka** [reply]
An editable table element is on our roadmap. We're not yet sure if it should be closer to a spreadsheet like element or a dataframe viewer (like ui.table) that is editable.

Do you want it to integrate with Pandas dataframes? Presumably edits made to the table should somehow be saved to disk?

**Reply by aforsythe** [reply]
Not sure what you mean "saved to disk". My primary use case is allowing input to a web app directly by editing the table.  Most would probably copy and paste data into the widget from excel. The data would be the input to an algorithm whose result would be displayed in the notebook.

**Reply by aforsythe** [reply]
@akshayka `mo.ui.experimental_data_editor` seems promising, but is there a way to paste data from an excel sheet into the the dataframe?  This is a critical feature for getting datasets into marimo quickly and easily.
